src/lib.rs

```diff
@@ -1324,43 +1352,52 @@ impl GooseAttack {
 
     // Configure how quickly to hatch Goose Users.
     fn set_hatch_rate(&mut self) -> Result<(), GooseError> {
+        // Track how value gets set so we can return a meaningful error if necessary.
+        let mut key = "configuration.hatch_rate";
+        let mut value = 0;
+
         // Check if --hash-rate is set.
         if let Some(hatch_rate) = self.configuration.hatch_rate {
-            // Setting --hatch-rate with --worker is not allowed.
-            if self.attack_mode == GooseMode::Worker {
-                return Err(GooseError::InvalidOption {
-                    option: "--hatch-rate".to_string(),
-                    value: hatch_rate.to_string(),
-                    detail:
-                        "The --hatch-rate option can not be set together with the --worker flag."
-                            .to_string(),
-                });
-            }
-
-            // Setting --hatch-rate of 0 is not allowed.
-            if hatch_rate == 0 {
-                return Err(GooseError::InvalidOption {
-                    option: "--hatch-rate".to_string(),
-                    value: "0".to_string(),
-                    detail: "The --hatch-rate option must be set to at least 1.".to_string(),
-                });
-            }
+            key = "--hatch_rate";
+            value = hatch_rate;
         // If not, check if a default hatch_rate is set.
         } else if let Some(default_hatch_rate) = self.defaults.hatch_rate {
             // On Worker hatch_rate comes from the Manager.
             if self.attack_mode == GooseMode::Worker {
                 self.configuration.hatch_rate = None;
             // Otherwise use default.
             } else {
+                key = "set_default(GooseDefault::HatchRate)";
+                value = default_hatch_rate;
                 self.configuration.hatch_rate = Some(default_hatch_rate);
             }
         // If not and if not running on Worker, default to 1.
         } else if self.attack_mode != GooseMode::Worker {
+            value = 1;
```

Comment:
I think this can never lead to an error.

Reply:
Agreed -- my thinking was more what happens if we make later code-changes and introduce an error.

Reply:
I've added the missing `key`, and documented why we set up debug here even though failure currently isn't possible.